UnityProject/Assets/Scripts/Items/Others/EmergencyOxygenTank.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using Objects.Atmospherics;
+using UnityEngine;
+
+namespace Items.Others
+{
+	public class EmergencyOxygenTank : MonoBehaviour, IPredictedInteractable<HandActivate>
+	{
+		private GasContainer container;
+		[SerializeField] private float gasReleaseOnPush = 10f;
+
+		private Vector3 rollbackMemory = Vector3.zero;
+
+		private void Awake()
+		{
+			container = GetComponent<GasContainer>();
+		}
+
+		public void ServerPerformInteraction(HandActivate interaction)
+		{
+			if(container.FullPercentage <= 1f) return;
+			Jetpack.PushPlayerInFacedDirection(interaction.PerformerPlayerScript, container, gasReleaseOnPush);
+		}
+
+		public void ClientPredictInteraction(HandActivate interaction)
+		{
+			if(container.FullPercentage <= 1f) return;
+			Jetpack.PushPlayerInFacedDirection(interaction.PerformerPlayerScript, container, gasReleaseOnPush);
+			Chat.AddExamineMsg(interaction.Performer, "You release some of the gas in this tank.");
+			rollbackMemory = interaction.Performer.RegisterTile().LocalPosition;
+		}
+
+		public void ServerRollbackClient(HandActivate interaction)
+		{
+			interaction.Performer.RegisterTile().ServerSetLocalPosition(rollbackMemory.CutToInt());
```

Comment:
rollbackMemory Is being set on client but then
is being used by server

just reset location on client There is a function for that

Reply:
done